internal/inventory/gcpfetcher/fetcher_assets.go

```diff
@@ -172,3 +194,173 @@ func appendIfExists(slice []string, fields map[string]*structpb.Value, key strin
 	}
 	return append(slice, value.GetStringValue())
 }
+
+func hasResourceData(item *gcpinventory.ExtendedGcpAsset) bool {
+	return item.Resource != nil && item.Resource.Data != nil
+}
+
+func getAssetTags(item *gcpinventory.ExtendedGcpAsset) []string {
+	if !hasResourceData(item) {
+		return nil
+	}
+
+	tagsObj, ok := item.GetResource().GetData().GetFields()["tags"]
+	if !ok {
+		return nil
+	}
+
+	structValue := tagsObj.GetStructValue()
+	if structValue == nil {
+		return nil
+	}
+
+	items, ok := structValue.GetFields()["items"]
+	if !ok {
+		return nil
+	}
+
+	tagValues := items.GetListValue().GetValues()
+	tags := make([]string, len(tagValues))
+	for i, tag := range tagValues {
+		tags[i] = tag.GetStringValue()
+	}
+
+	return tags
+}
+
+func getAssetLabels(item *gcpinventory.ExtendedGcpAsset) map[string]string {
+	if !hasResourceData(item) {
+		return nil
+	}
+
+	labels, ok := item.GetResource().GetData().GetFields()["labels"]
+	if !ok {
+		return nil
+	}
+
+	labelsMap := make(map[string]string)
+	if err := mapstructure.Decode(labels.GetStructValue().AsMap(), &labelsMap); err != nil {
+		return nil
+	}
+
+	return labelsMap
+}
+
+var assetEnrichers = map[string]func(item *gcpinventory.ExtendedGcpAsset, fields map[string]*structpb.Value) []inventory.AssetEnricher{
+	gcpinventory.IamRoleAssetType:               noopEnricher,
+	gcpinventory.CrmFolderAssetType:             noopEnricher,
+	gcpinventory.CrmProjectAssetType:            noopEnricher,
+	gcpinventory.StorageBucketAssetType:         noopEnricher,
+	gcpinventory.IamServiceAccountKeyAssetType:  noopEnricher,
+	gcpinventory.CloudRunService:                noopEnricher,
```

Comment:
cloud run has [applicable values](https://cloud.google.com/run/docs/reference/rest/v1/Container) we can collect, the issue is that they are provided in `asset.spec.template.spec.containers[].*`, so each value we can technically get (`name`, `image`) would be an array, as opposed to the expected single value

Reply:
in general, we could probably collect more ECS fields from each asset types if we dig deep enough

Reply:
Thanks for the clarification 👍